frontend/src/pages/SubmitPunchlinePage/index.tsx

```diff
@@ -10,34 +10,57 @@ import Setup from "../../components/Setup";
 import { useRound } from "../../contexts/round";
 import socket from "../../socket";
 import { usePlayers } from "../../contexts/players";
-import { usePunchlines } from "../../contexts/punchlines";
+import { PunchlinesAction, usePunchlines } from "../../contexts/punchlines";
+import { Punchline } from "../../types";
 
 const usePlayerIdState = createPersistedState("playerId");
 
 const SubmitPunchlinePage = ({ roundLimit }: { roundLimit: number }) => {
   const [, setResponse] = useState("");
 
   const { host, players } = usePlayers();
-  const { punchlines, removePunchline } = usePunchlines();
+  const [punchlines, dispatchPunchlines] = usePunchlines();
   const { roundNumber, setup, numPlayersChosen, incrementPlayersChosen } =
     useRound();
 
   const [playerId] = usePlayerIdState("");
   const playerIsHost = playerId === host;
 
-  const [punchlineSelected, setPunchlineSelected] = useState("");
-  const [punchlineSubmitted, setPunchlineSubmitted] = useState("");
+  const [punchlineSelected, setPunchlineSelected] = useState<Punchline>();
+  const [punchlineSubmitted, setPunchlineSubmitted] = useState<Punchline>();
 
-  const selectPunchline = (text: string) => {
+  const handleSelect = (punchline: Punchline) => {
     if (punchlineSubmitted) return;
 
-    if (text === punchlineSelected) {
-      setPunchlineSelected("");
+    if (punchline.text === punchlineSelected?.text) {
+      setPunchlineSelected(undefined);
     } else {
-      setPunchlineSelected(text);
+      setPunchlineSelected(punchline);
     }
   };
 
+  const handleSubmit = useCallback(() => {
+    if (punchlineSelected) {
+      // TODO: Remove this to rely on "punchlines:remove" event
+      dispatchPunchlines({
+        type: PunchlinesAction.REMOVE,
+        punchlines: [punchlineSelected.text],
+      });
+
+      setPunchlineSubmitted(punchlineSelected);
+      setPunchlineSelected(undefined);
+
+      incrementPlayersChosen();
+      socket.emit(
+        "round:player-choose",
+        [punchlineSelected.text],
+        (response: string) => {
+          setResponse(response);
+        }
+      );
```

Comment:
Again, slightly concerned that there's additional boilerplate/complexity being added

Reply:
This code already existed however was previously defined on the `<Button>` component in lines `96-113` (old) / `119` (new) - I moved it to the `handleSubmit` function since that's considered better practice.

Reply:
Right yeah, I guess I didn't see that in the PR